core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaMerchantOffersTranslator.java

```diff
@@ -144,12 +146,31 @@ public static void openMerchant(GeyserSession session, ClientboundMerchantOffers
         session.sendUpstreamPacket(updateTradePacket);
     }
 
-    private static NbtMap getItemTag(GeyserSession session, ItemStack stack, int specialPrice) {
-        ItemData itemData = ItemTranslator.translateToBedrock(session, stack);
+    private static NbtMap getItemTag(GeyserSession session, ItemStack stack) {
+        if (stack == null || stack.getAmount() <= 0) { // Negative item counts appear as air on Java
+            return NbtMap.EMPTY;
+        }
+        return getItemTag(session, stack, session.getItemMappings().getMapping(stack), stack.getAmount());
+    }
+
+    private static NbtMap getItemTag(GeyserSession session, ItemStack stack, int specialPrice, int demand, float priceMultiplier) {
+        if (stack.getAmount() <= 0) { // Negative item counts appear as air on Java
```

Comment:
Does this also need a null check?

Reply:
I've just checked, this takes the first input item of the trade which seems to be incorrectly annotated as `@NonNull` in MCProtocolLib.

Reply:
I'm pretty sure any non-null-annotated ItemStack in MCProtocolLib is technically incorrect.